openstack/baremetal/v1/nodes/requests.go

```diff
@@ -438,3 +438,49 @@ func ChangeProvisionState(client *gophercloud.ServiceClient, id string, opts Pro
 	})
 	return
 }
+
+type TargetPowerState string
+
+// TargetPowerState is used when changing the power state of a node.
+const (
+	PowerOn       TargetPowerState = "power on"
+	PowerOff                       = "power off"
+	Rebooting                      = "rebooting"
+	SoftPowerOff                   = "soft power off"
+	SoftRebooting                  = "soft rebooting"
+)
+
+// PowerStateOptsBuilder allows extensions to add additional parameters to the ChangePowerState request.
+type PowerStateOptsBuilder interface {
+	ToPowerStateMap() (map[string]interface{}, error)
+}
+
+// PowerStateOpts for a request to change a node's power state.
+type PowerStateOpts struct {
+	Target  TargetPowerState `json:"target" required:"true"`
+	Timeout int              `json:"timeout,omitempty"`
```

Comment:
One more question: will there ever be a need to specify a timeout of 0? If so, then this will need to be changed to `*int`.

Reply:
Ah good question - it cannot be 0, so I think this is good to go!

https://github.com/openstack/ironic/blob/master/ironic/api/controllers/v1/node.py#L505-L506